### PR TITLE
Feature/evaluate expression refractor

### LINKED ISSUE #370: Static evaluation bailing out from re-exports
**Current behaviour**

```js
// colors.js
export const primary = '#8777D9';
```

```js
// index.js
import {
  primary,
} from './colors';

export const colors = {
  primary,
};
```

```jsx
// my-component.js
import '@compiled/react';
import { colors } from './index';

<div css={{ color: colors.primary }} /> // --_6oohme: #8777D9;
```

**Expected behaviour**

Should be inlined and not a CSS variable.

### PR DESCRIPTION
**Pull request by at-nathan**
Resolves #370, #357

A refactor of `evaluate-expression.tsx` and rework of how we currently evaluate member expressions. Some key points:

1. Move `./evaluate-expression.tsx` -> `./evaluate-expression/index.tsx`. To allow code splitting of functionality
2. Recursively evaluates member expressions.  Resolves:
- https://github.com/atlassian-labs/compiled/issues/370
- https://github.com/atlassian-labs/compiled/issues/357
- Handles mix expressions in member like: `theme.getColor('dark').color`
- Handles nested namespace imports
3. Implement modular code split so we can have more granular unit tests

### CODE REVIEW

File: packages/babel-plugin/src/utils/evaluate-expression/index.tsx [hunk 1]
```diff
@@ -566,7 +405,11 @@ export const evaluateExpression = (
     // It's preferable to use the identifier than its result if it can't be statically evaluated.
     // E.g. say we got the result of an identifier `foo` as `bar()` -- its more preferable to return
     // `foo` instead of `bar()` for a single source of truth.
-    const babelEvaluatedNode = babelEvaluateExpression(value, updatedMeta, expression);
+    const babelEvaluatedNode = babelEvaluateExpression(
+      value as t.Expression,
```
Comment (at-nathan) [inline packages/babel-plugin/src/utils/evaluate-expression/index.tsx:409]: hmm got `value as t.Expression` back as part of a bad merge. Can probably remove the `as t.Expression`

File: .changeset/small-swans-approve.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@compiled/babel-plugin': minor
+---
+
+Add support for namespace imports and export specifiers
```
Comment (at-nathan) [inline .changeset/small-swans-approve.md:5]: Not sure how this got in, either way should remove this and re-generate

File: packages/babel-plugin/src/utils/evaluate-expression/index.tsx [hunk 3]
```diff
@@ -40,158 +36,6 @@ const traverseIdentifier = (expression: t.Identifier, meta: Metadata) => {
   return createResultPair(value as t.Expression, updatedMeta);
 };
 
-/**
```
Comment (at-nathan) [inline packages/babel-plugin/src/utils/evaluate-expression/index.tsx:43]: All this removed code was handling member expression evaluation. I've tried to simplify it into `traverseMemberAccessPath`

File: packages/babel-plugin/src/__tests__/module-traversal.test.tsx [hunk 4]
```diff
@@ -530,18 +574,10 @@ describe('module traversal', () => {
...
-    `);
+        <div css={{ ...objects.colorMixin2(color.blue) }} />
+      `);
 
-      expect(result).toIncludeMultiple([
-        '{color:red}',
-        // We're not handling statically evaluating param values used from a namespace chain,
-        // as the long term strategy is to only statically evaluate expressions scoped locally
-        // to the current file.
-        '{background-color:var(--_1w9snyi)}',
```
Comment (at-nathan) [inline packages/babel-plugin/src/__tests__/module-traversal.test.tsx:543]: Was able evaluate this statically by treating call expressions like `objects.colorMixin2(color.blue)` as a member expression

File: packages/babel-plugin/src/utils/traversers/types.tsx [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import type { NodePath } from '@babel/traverse';
+import type * as t from '@babel/types';
+
+export type Result<T> = {
+  node: t.Node;
+  path: NodePath<T>;
```
[the remark this answers is not in the document]
Comment (at-nathan) [reply packages/babel-plugin/src/utils/traversers/types.tsx:6]: The node we return isn't `path.node` for example in object we return `path.node.value`. And in exports it's deeply nest and involves an array 😓

**Comment by madou** [comment]
Nathan what review are you looking for? Anything I should focus on?

**Comment by at-nathan** [comment]
@madou just making sure nothing I'm doing is too controversial. Like the the way I'm evaluating member expressions now, and the whole code split taking `evaluate-expression.tsx` -> `evaluate-expression/index.tsx`

File: packages/babel-plugin/src/__tests__/module-traversal.test.tsx [hunk 6]
```diff
@@ -222,6 +222,21 @@ describe('module traversal', () => {
     expect(result).toInclude('{color:orange}');
   });
 
+  it('should inline css from a member expression that comprises of an import being exposed by a local variable', () => {
+    const result = transform(
+      `
```
[the remark this answers is not in the document]
Comment (at-nathan) [reply packages/babel-plugin/src/__tests__/module-traversal.test.tsx:227]: hmm I copied this format from the rest of the tests in this file... Maybe time for some prettier rules on this format?

**Review by JakeLane — DISMISSED** [review]

**Review by MonicaOlejniczak — APPROVED** [review]

**Pull request merged**